rfd/XXXX-tsh-upgrade.md

```diff
@@ -0,0 +1,117 @@
+Teleport documents several different methods for installing the client tools. https://goteleport.com/docs/installation/
+- Teleport can be installed from source and built using go. This scenario can be handled the same as if Teleport was installed by downloading the pre-build binaries.
+- Teleport pre-built binaries can be downloaded for the desired platform.
+- Teleport can be installed through the system package managers. Teleport currently maintains DEB and RPM repositories.
+- Teleport is available for installation using brew on macOS, but it is not officially supported by Teleport and not recommended.
+- Teleport provides macOS installation packages. One package installs all Teleport binaries, while the other installs only the tsh client.
+- Teleport provides a tsh only zip installation for windows.
+
+Additionally, Teleport supports a couple different releases: Open Source, Enterprise, and a FIPS compliant release.
+The upgrade process will need to detect which release was installed and make sure to re-install the same release.
+
+In order to identify which installation method was used to install the Teleport client tools, various checks will be executed.
+The default installation method will be downloading the pre-built binaries.
+
+#### Linux
+
+Teleport currently maintains DEB and RPM repositores.
+So before upgrading, the upgrade command needs to identify whether the system it's running on is a Debian or Red Hat distribution.
+The upgrade command will look at the /etc/lsb-release or /etc/redhat-release files to identify the Linux distribution and to determine which package manager to use.
+
+The installation method can be further narrowed down by using the package manager CLI tools dpkg for Debian, and rpm for Red Hat.
+If Teleport was installed using the manager, dpkg or rpm can be used to identify which release was installed: teleport, teleport-ent, or teleport-ent-fips.
+Once the release has been identified, the Teleport client tools can now be upgraded using the package manager.
+
+If Teleport was not installed using the package manager, the Teleport client tools will be upgraded by downloading the pre-built binaries.
+
+#### macOS
+
+On macOS, the Teleport package is available for installation using the macOS package installer.
+To detect whether or not Teleport was installed using the installer, the pkgutil command can be used to search for the Teleport package.
+Only Teleport Open Source and Teleport Enterprise are supported on macOS.
+
+A signed release of tsh is also supported on macOS and is required to use TouchID.
+So tsh will need to be upgraded separately after the initial upgrade of the other Teleport client tools.
+
+#### Windows
+
+Only the tsh command is supported on Windows machines.
+tsh only supports an Open Source release, so for windows machines, the tsh upgrade command simply needs to download the desired tsh binary and install it.
+
+## UX
```

Comment:
Can you also describe the following things we discussed:

- Will the users be able to disable the auto-upgrade behavior? How? Describe [tsh config](https://goteleport.com/docs/reference/cli/#tsh-configuration-files) changes.
- What will the default behavior be? We should default to auto-upgrade.